### DISCUSSION TITLE
Table: Trigger Modal from Action in Another Modal

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#6185] [Help]
I have the following action code in a table, that opens a modal:

```php
Tables\Actions\Action::make('profile')
    ->label('Profile')
    ->color('success')
    ->action(function (array $data): void {
        // I want to close this modal and open another one
    })
    ->modalContent(fn ($record) => view('filament.modals.profile', ['record' => $record]))
```

When the user clicks the submit button, I'd like to close that modal and open this modal from another action:

```php
Tables\Actions\Action::make('message')
    ->label('Message')
    ->action(function (Registrant $record, array $data): void {
        // action logic here
    })
    ->form([
        Forms\Components\Textarea::make('body')
            ->label('')
            ->required(),
    ])
    ->hidden(),
```

Is that possible?

**Comment by danharrin** [comment]
Currently modals cannot be nested

**Reply by danharrin** [reply]
Maybe you want a wizard

**Reply by intrepidws** [reply]
I don't want to nest it really - just hoping for another way to trigger it. Both modals exist as actions on the table.

**Reply by danharrin** [reply]
You can try `$livewire->mountTableAction('message', $record)`

**Reply by intrepidws** [reply]
No dice. Thanks for trying.

**Comment by ArtMin96** [comment]
This is just a suggestive option. For Filament pages, this might help you. Call this when the first action is finished.

```php
$this->displayMountedAction('message', ['body' => 'Some text']);
```

```php
trait CanCallMountedAction
{
    public function displayMountedAction(string $action, ?array $data = []): void
    {
        $this->mountAction($action);

        if ($data) {
            $this->getMountedActionForm()->fill($data);
        }
    }
}
```

**Reply by intrepidws** [reply]
Thanks for trying but I can't get that to work either.

**Reply by tomsgad** [reply]
Any luck with this? Got stuck with the same issue!

**Comment by divdax** [comment]
I try to render an action button within a view that is loaded by an action. Although the button renders successfully, clicking on it does not open the modal. Do you have any ideas on how to make this work?

**Comment by ismail17719** [comment]
Hello @danharrin, I need to trigger one modal based on the action of a previous modal. That is the kind of scenario which you refer to as nested modals. I also read your comment about Filament not supporting nested modals. Is there a workaround this?

**Reply by leandrocfe** [reply]
Maybe `replaceMountedAction`
https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component#chaining-actions

**Reply by danharrin** [reply]
Filament supports nested modals in v3 too, my comment is probably old

**Reply by ismail17719** [reply]
Thank you @leandrocfe & @danharrin I was able to achieve the desired effect with your suggestions and updates.

**Reply by T3mmyKay** [reply]
$this->replaceMountedAction I don't think it works for table actions

**Reply by VuongPham2001** [reply]
i have used $this->replaceMountedAction in EditAction and it don't work. Does anyone know how to do this? I'm so stuck 

@danharrin do you have any idea for this
